Assets/Scripts/UI/ContextualHotKeysUI.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ContextualHotKeysUI : MonoBehaviour
+{
+    public Transform scrollViewContent;
+    public GameObject hotKeyPrefab;
+
+    [SerializeField] private Button closeButton;
+
+    Dictionary<string, TMP_Text> buttonKeyCodeTexts;
+
+    private HotKeyManager _hotKeyManager;
+
+    private HashSet<string> terrainHotKeys = new HashSet<string> { HotkeyConstants.SelectPosition, HotkeyConstants.SelectRotation, HotkeyConstants.SelectScale };
+
+    private HashSet<string> propHotKeys = new HashSet<string> { HotkeyConstants.DeletePrefab, HotkeyConstants.Focus, HotkeyConstants.Duplicate };
+
+
+    private void Awake()
+    {
+        _hotKeyManager = HotKeyManager.GetInstance();
+
+        UIManager.OnEditModeChanged += PopulateHotkeysBasedOnGameMode;
+
+        closeButton.onClick.RemoveAllListeners();
+        closeButton.onClick.AddListener(ToggleUI);
+    }
+
+    private void PopulateHotkeysBasedOnGameMode(EditMode editMode)
+    {
+        CleanScrollView();
+
+        Dictionary<string, KeyCode> keys = _hotKeyManager.GetHotKeys();
+        Dictionary<string, string> tooltips = _hotKeyManager.GetTooltips();
+
+        HashSet<string> editModeKeys;
+
+        if (editMode == EditMode.Terrain)
+            editModeKeys = terrainHotKeys;
+        else
+            editModeKeys = propHotKeys;
+
+
+        buttonKeyCodeTexts = new Dictionary<string, TMP_Text>();
+        foreach (string hotkey in keys.Keys)
+        {
+            if (hotkey == HotkeyConstants.ModifierKey)
+                return;

```

Comment:
do you mean continue and not return?

Reply:
I can't figure out why I did this. I just removed:
`if (hotkey == HotkeyConstants.ModifierKey)
                return;`

Reply:
you likely did it so that control doesnt show up

Reply:
add comments for these things in the future :)